PORTAL/jobs/_pyperformance.py

```diff
@@ -1497,14 +1498,10 @@ def uploadid(self) -> PyperfUploadID:
         try:
             return self._uploadid
         except AttributeError:
-            if self._resfile.uploadid:
-                # XXX Compare with what we get from the metadata?
-                self._uploadid = self._resfile.uploadid
-            else:
-                self._uploadid = PyperfUploadID.from_metadata(
-                    self.metadata,
-                    suite=self.suites,
-                )
+            self._uploadid = PyperfUploadID.from_metadata(
+                self.metadata,
+                suite=self.suites,
+            )
```

Comment:
It would be worth verifying that the original results file's upload ID (derived from the filename) matches the metadata, but only if the results are unmodified and only if the results file has an uploadid.

```suggestion
            self._uploadid = PyperfUploadID.from_metadata(
                self.metadata,
                suite=self.suites,
            )
            assert (self._modified or
                       self._resfile.uploadid is None or
                       self._uploadid == self._resfile.uploadid), (self._uploadid, self._resfile.uploadid)
```

That check represents an expected invariant, hence an assert instead of raising `ValueError`.

Reply:
Yeah, unfortunately, this does fail, which is why I thought -- by the time we've computed the metadata, that's actually "correct" and should override whatever is in `_resfile.uploadid` anyway...  

It fails when we load the input file, before we split it...  That doesn't / shouldn't have the `_modified` bit set..., it's just that the suite from the filename is "None", but when it reads the metadata it's `['pyperformance', 'pyston']`...